### ISSUE TITLE
node panic when connected with websocket

### ISSUE THREAD

**Issue by benbaley** [PlatONnetwork/PlatON-Go#1757]
Hi there,

please note that this is an issue tracker reserved for bug reports and feature requests.

#### System information

PlatON version: f2d7ad2fdb16774478fd7cfc13f4e76cd82cc162
OS & Version: Windows/Linux/OSX
Commit hash : f2d7ad2fdb16774478fd7cfc13f4e76cd82cc162

#### Expected behaviour
success

#### Actual behaviour
node panic

#### Steps to reproduce the behaviour
1. start node and wait new block generated
2. connect to node by ws

#### Backtrace

````
DEBUG[04-06|11:17:41.621|internal/ethapi/api.go:668]                Executing VM call finished               runtime=165.042µs
panic: runtime error: slice bounds out of range [32:14]

goroutine 9249 [running]:
github.com/PlatONnetwork/PlatON-Go/x/xutil.IsWorker(...)
	/opt/jenkins/workspace/PlatON/build_ubuntu/x/xutil/calculate.go:46
github.com/PlatONnetwork/PlatON-Go/core.NewEVMContext(0x160bf30, 0xc01955e780, 0xc0002e4500, 0x15f3ba0, 0xc00036c000, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/opt/jenkins/workspace/PlatON/build_ubuntu/core/evm.go:47 +0x8e5
github.com/PlatONnetwork/PlatON-Go/eth.(*EthAPIBackend).GetEVM(0xc000337280, 0x1601888, 0xc01955e720, 0x160bf30, 0xc01955e780, 0xc0000e0a00, 0xc0002e4500, 0x0, 0x5c, 0x0, ...)
	/opt/jenkins/workspace/PlatON/build_ubuntu/eth/api_backend.go:142 +0x93
github.com/PlatONnetwork/PlatON-Go/internal/ethapi.(*PublicBlockChainAPI).doCall(0xc000196e30, 0x1601888, 0xc01955e720, 0x10, 0x0, 0x1000000, 0xc01952c708, 0x0, 0x0, 0x0, ...)
	/opt/jenkins/workspace/PlatON/build_ubuntu/internal/ethapi/api.go:708 +0x4d2
github.com/PlatONnetwork/PlatON-Go/internal/ethapi.(*PublicBlockChainAPI).Call(0xc000196e30, 0x16018c0, 0xc019917620, 0x10, 0x0, 0x1000000, 0xc01952c708, 0x0, 0x0, 0x0, ...)
	/opt/jenkins/workspace/PlatON/build_ubuntu/internal/ethapi/api.go:740 +0x9c
reflect.Value.call(0xc00042a780, 0xc000116e30, 0x13, 0x13fbd6e, 0x4, 0xc01955e6c0, 0x4, 0x4, 0xc01955e6f0, 0xc0197b4fc0, ...)
	/usr/lib/go-1.16/src/reflect/value.go:476 +0x8e7
reflect.Value.Call(0xc00042a780, 0xc000116e30, 0x13, 0xc01955e6c0, 0x4, 0x4, 0x2, 0x4, 0xc0003318f0)
	/usr/lib/go-1.16/src/reflect/value.go:337 +0xb9
github.com/PlatONnetwork/PlatON-Go/rpc.(*Server).handle(0xc018e420f0, 0x16018c0, 0xc019917620, 0x160c638, 0xc019248fa0, 0xc01955e660, 0x0, 0xc0003b2f60, 0xa7eb7d)
	/opt/jenkins/workspace/PlatON/build_ubuntu/rpc/server.go:310 +0x69b
github.com/PlatONnetwork/PlatON-Go/rpc.(*Server).exec(0xc018e420f0, 0x16018c0, 0xc019917620, 0x160c638, 0xc019248fa0, 0xc01955e660)
	/opt/jenkins/workspace/PlatON/build_ubuntu/rpc/server.go:345 +0x1ca
github.com/PlatONnetwork/PlatON-Go/rpc.(*Server).serveRequest.func2(0xc0198753f0, 0xc018e420f0, 0xc0196891f0, 0x160c638, 0xc019248fa0, 0xc000116e80, 0x1, 0x1, 0x0)
	/opt/jenkins/workspace/PlatON/build_ubuntu/rpc/server.go:205 +0xb4
created by github.com/PlatONnetwork/PlatON-Go/rpc.(*Server).serveRequest
	/opt/jenkins/workspace/PlatON/build_ubuntu/rpc/server.go:200 +0x337
````

**Comment by benbaley** [comment]
When a call is initiated through websocket, a judgment error occurs when getting state of current block number  The `extra` field length of the genesis block is not considered to be less than 32 bytes, which leads to the out-of-bounds array and the node Panic.
this bug had fixed in #1755

**Comment by benbaley** [comment]
fixed
